release/nightly_tests/stress_tests/test_threaded_actors.py

```diff
@@ -0,0 +1,163 @@
+import argparse
+import json
+import logging
+import numpy as np
+import os
+import threading
+import time
+
+from ray._private.test_utils import monitor_memory_usage
+from ray.data.impl.progress_bar import ProgressBar
+
+from collections import namedtuple
+from queue import Queue
+
+import ray
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+PiResult = namedtuple("PiResult", ["samples", "pi"])
+
+
+@ray.remote(num_cpus=1)
+class PiCalculator:
+    def __init__(self, metadata):
+        # -- Read only variables --
+        self.metadata = metadata
+        self.sample_batch = 1000000
+        # -- Variables that are accessed by mulitple threads --
+        self.lock = threading.Lock()
+        self.result_queue = Queue()
+        self.is_running = False
+
+    def ready(self):
+        pass
+
+    def run_compute(self):
+        self.is_running = True
+        sample_cnt = 0
+        while self.is_running:
+            # Compute pi
+            xs = np.random.uniform(low=-1.0, high=1.0, size=self.sample_batch)
+            ys = np.random.uniform(low=-1.0, high=1.0, size=self.sample_batch)
+            xys = np.stack((xs, ys), axis=-1)
+            inside = xs * xs + ys * ys <= 1.0
+            xys_inside = xys[inside]
+            in_circle = xys_inside.shape[0]
+            approx_pi = 4.0 * in_circle / self.sample_batch
+
+            # Put the result to the queue.
+            sample_cnt += self.sample_batch
+            with self.lock:
+                self.result_queue.put(
+                    PiResult(samples=sample_cnt, pi=approx_pi))
+
+    def stop(self):
+        self.is_running = False
+
+    def get_metadata(self):
+        return self.metadata
+
+    def get_pi(self):
+        result = None
+        while not result:
+            with self.lock:
+                if not self.result_queue.empty():
+                    result = self.result_queue.get(block=False)
+                time.sleep(1)
+        return result
+
+
+def start_actors(num_actors, num_nodes):
+    """Create actors and run the computation loop.
+    """
+    num_actors = int(num_actors)
+    start = time.time()
+    nodes = []
+    # Place an actor per node in round-robin.
+    # It is added here to simulate the real user workload.
+    while len(nodes) < num_nodes:
+        nodes = [
+            next((r for r in n["Resources"] if "node" in r), None)
+            for n in ray.nodes() if n["Alive"]
+        ]
+        nodes = [n for n in nodes if n is not None]
+    pi_actors = [
+        PiCalculator.options(resources={
+            n: 0.01
+        }, max_concurrency=10).remote({
+            "meta": 1
+        }) for n in nodes for _ in range(num_actors)
```

Comment:
I think this starts `num_actors * num_nodes` actors, where num_actors is total number of cpus in cluster?

Reply:
here num_actors == num_cpus per node * num_nodes! So you are right